### DISCUSSION TITLE
stuck with a weird gcs caching issue, maybe osx-arm related?

### DISCUSSION THREAD

**Discussion by willie** [microsoft/vcpkg#31952] [Q&A] [answered]
I've got GCS caching working great on a number of targets, but not all of them. Here is the setup:  `VCPKG_BINARY_SOURCES="x-gcs,gs://BUCKETNAME/,readwrite"`

Intel iMac (Monterey): docker ubuntu, mac ninja and xcode build 
M1 MacBook (Ventura): docker ubuntu build, **mac ninja and mac xcode builds cannot see the cache**

The specific error is:

> ServiceException: 401 Anonymous caller does not have storage.objects.get access to the Google Cloud Storage object. Permission 'storage.objects.get' denied on resource (or it may not exist).

On the same machine (M1), from the same Terminal, I can do `gsutil ls gs://BUCKETNAME` and it works. The docker ubuntu builds that work are using a copy of the ` ~/.config/gcloud` directory for authentication.

I'm baffled.

**Comment by willie** [comment, marked as answer]
OK, so this is super weird. I decided to see if there were rogue copies of `gsutil` lingering around on my drive. For some reason, there was this: `./vcpkg/downloads/tools/gsutil-4.65-osx`. I removed that directory and ran the build again and it seems to be working now. I have no idea why vcpkg installed an old gsutil or why it decided to keep using 4.65 of gsutil when I have 5.24 installed otherwise.

**Reply by dg0yt** [reply]
This tool and version is listed in `scripts/vcpkgTools.xml`.
